telethon/network/connection/tcpabridged.py

```diff
@@ -9,34 +9,36 @@ class ConnectionTcpAbridged(Connection):
     only require 1 byte if the packet length is less than
     508 bytes (127 << 2, which is very common).
     """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._codec = AbridgedPacket()
+
     def _init_conn(self):
-        self._writer.write(b'\xef')
+        self._writer.write(self._codec.tag)
 
-    def _write(self, data):
-        """
-        Define wrapper write methods for `TcpObfuscated` to override.
-        """
-        self._writer.write(data)
+    def _send(self, data):
+        self._writer.write(self._codec.encode_packet(data))
 
-    async def _read(self, n):
-        """
-        Define wrapper read methods for `TcpObfuscated` to override.
-        """
-        return await self._reader.readexactly(n)
+    async def _recv(self):
+        return await self._codec.read_packet(self._reader)
 
-    def _send(self, data):
+
+class AbridgedPacket:
```

Comment:
Maybe `AbridgedCodec` is a better name?

Reply:
Sure. I was just thinking that `codec` is too generic, but all these `abridged` `full` `intermediate` are used to encode data packets. So, maybe `AbridgedPacketCodec` might be a better name?

Reply:
But, on the other side, `Abridged` etc is not used for anything else than codecs, so, maybe we don't need it to be too specific.

Reply:
I personally would prefer the `packet` instead of `IO` and `codec` for plain / obfuscated.